src/main/java/me/jellysquid/mods/sodium/common/util/PathUtil.java

```diff
@@ -0,0 +1,21 @@
+package me.jellysquid.mods.sodium.common.util;
+
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class PathUtil {
+    /**
+     * Resolves a timestamped sibling of the base path - {@code "{base_dir}/{base_name}_{tag}_{timestamp}.{base_ext}"}.
+     * @param base Base path
+     * @param tag Tag to add to sibling's name
+     * @return Sibling path
+     */
+    public static Path resolveTimestampedSibling(Path base, String tag) {
+        LocalDateTime now = LocalDateTime.now();
+        String backupName = base.getFileName().toString();
+        backupName = backupName.substring(0, backupName.lastIndexOf('.') - 1);
```

Comment:
~This is not a reliable way to determine the file name without extension. In fact, this website's first example of how not to ask a technical question uses file extension handling to convey the point: https://xyproblem.info/~

Nevermind, it appears the Microsoft .NET standard library uses the "last `.` truncation" technique: [https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?redirectedfrom=MSDN&view=net-5.0#System_IO_Path_GetFileNameWithoutExtension_System_String_](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?redirectedfrom=MSDN&view=net-5.0#System_IO_Path_GetFileNameWithoutExtension_System_String_)

Reply:
@Leo40Git Hey, turns out a relevant PR in OpenJDK was opened recently! https://github.com/openjdk/jdk/pull/2319